jgrapht-core/src/main/java/org/jgrapht/alg/shortestpath/ALTAdmissibleHeuristic.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * (C) Copyright 2016-2016, by Dimitrios Michail and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.alg.shortestpath;
+
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+import org.jgrapht.DirectedGraph;
+import org.jgrapht.Graph;
+import org.jgrapht.UndirectedGraph;
+import org.jgrapht.alg.interfaces.AStarAdmissibleHeuristic;
+import org.jgrapht.alg.interfaces.ShortestPathAlgorithm.SingleSourcePaths;
+import org.jgrapht.alg.util.ToleranceDoubleComparator;
+import org.jgrapht.graph.EdgeReversedGraph;
+
+/**
+ * An admissible heuristic for the A* algorithm using a set of landmarks and the triangle
+ * inequality. Assumes that the graph contains non-negative edge weights. Preprocessing requires two
+ * shortest path computations per landmark using Dijkstra's algorithm. The heuristic's space
+ * requirement is O(n) per landmark where n is the number of vertices of the graph.
+ * 
+ * <p>
+ * The method generally abbreviated as ALT (from A*, Landmarks and Triangle inequality) is described
```

Comment:
Could you add a brief description of the method and how it works here (just a few lines summarizing the approach)?